### ISSUE TITLE
Importing nest in python mutates sys.argv

### ISSUE THREAD

**Issue by maharjun** [nest/nest-simulator#779]
The following lines of code

```
echo "import sys; print(sys.argv)" > temppy.py
python temppy.py testarg
```

results in the the following output
```
['temppy.py', 'testarg']
```

which is as expected. However, the following lines of code:

```
echo "import sys; import nest; print(sys.argv)" > temppy.py
python temppy.py testarg
```

result in the following:
```
[INFO] [2017.7.5 17:47:44 /home/arjun/RandomRepos/nest-simulator/nestkernel/rng_manager.cpp:238 @ Network::create_rngs_] : Creating default RNGs
[INFO] [2017.7.5 17:47:44 /home/arjun/RandomRepos/nest-simulator/nestkernel/rng_manager.cpp:233 @ Network::create_rngs_] : Deleting existing random number generators
[INFO] [2017.7.5 17:47:44 /home/arjun/RandomRepos/nest-simulator/nestkernel/rng_manager.cpp:238 @ Network::create_rngs_] : Creating default RNGs
[INFO] [2017.7.5 17:47:44 /home/arjun/RandomRepos/nest-simulator/nestkernel/rng_manager.cpp:284 @ Network::create_grng_] : Creating new default global RNG

              -- N E S T --

  Copyright (C) 2004 The NEST Initiative
  Version 2.12.0 Jul  5 2017 11:36:46

This program is provided AS IS and comes with
NO WARRANTY. See the file LICENSE for details.

Problems or suggestions?
  Visit http://www.nest-simulator.org

Type 'nest.help()' to find out more about NEST.
['pynest', 'testarg']
```

Here we clearly see that the importing of the nest module results in `sys.argv[0]` being changed to `pynest`

COMMIT ID: a721e62116fd57349439af4cf863835d596918b3
Cmake flags: `-Dwith-python=3 -Dwith-mpi=OFF -Dwith-optimize=ON -Dwith-debug=ON ../../nest-simulator/`

**Comment by terhorstd** [comment]
That is very interesting. Is there a particular problem with this you need to have solved?

**Comment by maharjun** [comment]
When using python, it is useful in many cases to have access to the command used to execute it (for example, I would like to store the command used to execute a run so that I can reproduce it in the future). If the name of the main file (`sys.argv[0]`) is overwritten then this becomes impossible.

My current workaround is to create a shim from which I import nest. In the shim I store and reassign the previous value of `sys.argv[0]`. However, I think it is highly non-standard to modify `sys.argv` implicitly on model loading, and so I think that this behaviour needs to be fixed.

**Comment by heplesser** [comment]
This is a recurring issue, see also #163, #164, #377, #393. I had thought this would have been fixed by #393, but apparently not. @apeyser Could you take a look?
